api/iot/v1beta1/iot_helpers.go

```diff
@@ -0,0 +1,56 @@
+package iot
+
+import (
+	"time"
+
+	"github.com/scaleway/scaleway-sdk-go/internal/async"
+	"github.com/scaleway/scaleway-sdk-go/internal/errors"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+const (
+	waitForHubDefaultTimeout = 15 * time.Minute
+	defaultRetryInterval     = 5 * time.Second
+)
+
+// WaitForHubRequest is used by WaitForHub method.
+type WaitForHubRequest struct {
+	HubID         string
+	Region        scw.Region
+	Status        HubStatus
+	Timeout       *time.Duration
+	RetryInterval *time.Duration
+}
+
+// WaitForHub waits for the hub to be in a ready state before returning.
+func (s *API) WaitForHub(req *WaitForHubRequest) (*Hub, error) {
+	timeout := waitForHubDefaultTimeout
+	if req.Timeout != nil {
+		timeout = *req.Timeout
+	}
+	retryInterval := defaultRetryInterval
+	if req.RetryInterval != nil {
+		retryInterval = *req.RetryInterval
+	}
+
+	hub, err := async.WaitSync(&async.WaitSyncConfig{
+		Get: func() (interface{}, bool, error) {
+			hub, err := s.GetHub(&GetHubRequest{
+				HubID:  req.HubID,
+				Region: req.Region,
+			})
+			if err != nil {
+				return nil, false, err
+			}
+
+			return hub, hub.Status == HubStatusReady, nil
```

Comment:
What is hub is in error or lock or disabled, these are final status too ?  Have a look at instance WaitServer on how we manage multiple terminal status

Reply:
I updated to add these transient statuses too. For lock, this is not returned as a state but as an error in cases of writes on a locked hub.